File: Swift/백준/Silver/10845. 큐/큐.swift
import Foundation

struct Queue {
    var queue: [Int] = []
    private var front: Int = 0
    
    mutating func push(_ value: Int) {
        queue.append(value)
    }
    
    mutating func pop() -> Int {
        guard let item = queue.first else { return -1 }
        queue.removeFirst()
        return item
    }
    
    mutating func size() -> Int {
        return queue.count
    }
    
    mutating func empty() -> Int {
        return queue.isEmpty ? 1 : 0
    }
    
    mutating func frontValue() -> Int {
        return queue.first ?? -1
    }
    
    mutating func backValue() -> Int {
        return queue.last ?? -1
    }
}

let inputNumber = Int(readLine()!)!


var resultQueue: Queue = Queue()

for _ in 0..<inputNumber {
    let inputArr = readLine()!.split(separator: " ")
    switch inputArr[0] {
    case "push":
        resultQueue.push(Int(inputArr[1])!)
    case "pop":
        print(resultQueue.pop())
    case "size":
        print(resultQueue.size())
    case "empty":
        print(resultQueue.empty())
    case "front":
        print(resultQueue.frontValue())
    case "back":
        print(resultQueue.backValue())
    default:
        break
    }
}

File: Java/백준/Silver/10845. 큐/큐.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        ArrayDeque<Integer> ad = new ArrayDeque<>();
        int count = sc.nextInt();

        for (int i = 0; i<=count; i++) {
            String[] arr = sc.nextLine().split(" ");
            String str = arr[0];
            if (str.equals("push")) {
                ad.addLast(Integer.parseInt(arr[1]));
            } else if (str.equals("pop")) {
                if (ad.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(ad.pollFirst());
                }
            } else if (str.equals("size")) {
                System.out.println(ad.size());
            } else if (str.equals("empty")) {
                System.out.println(ad.isEmpty() ? 1 : 0);
            } else if (str.equals("front")) {
                if (ad.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(ad.peekFirst());
                }
            } else if (str.equals("back")) {
                if (ad.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(ad.peekLast());
                }
            }
        }
    }
}